apps/server/src/modules/learnroom/uc/course-copy.uc.ts

```diff
@@ -0,0 +1,34 @@
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { Actions, CopyStatusDTO, EntityId, Permission } from '@shared/domain';
+import { CourseCopyService } from '@shared/domain/service/course-copy.service';
+import { AuthorizationService } from '@src/modules/authorization/authorization.service';
+import { CourseRepo } from '../../../shared/repo';
+
+@Injectable()
+export class CourseCopyUC {
+	constructor(
+		private readonly courseRepo: CourseRepo,
+		private readonly authorisation: AuthorizationService,
+		private readonly courseCopyService: CourseCopyService
+	) {}
+
+	async copyCourse(userId: EntityId, courseId: EntityId): Promise<CopyStatusDTO> {
+		const user = await this.authorisation.getUserWithPermissions(userId);
+		const originalCourse = await this.courseRepo.findById(courseId);
+		if (
+			!this.authorisation.hasPermission(user, originalCourse, {
+				action: Actions.write,
+				requiredPermissions: [Permission.COURSE_CREATE],
+			})
+		) {
+			throw new ForbiddenException();
+		}
+		const { copy, status } = this.courseCopyService.copyCourseMetadata({
+			originalCourse,
+			user,
+		});
+		await this.courseRepo.save(copy);
+		status.id = copy.id;
```

Comment:
not quite happy with this line as it is very close to being logic, but since it will naturally disappear in a future iteration this is fine for right now.